### PR TITLE
feat(clipboard-copy): add clipboard-copy

### PR DESCRIPTION
**Pull request by heyMP**
Issue: https://github.com/patternfly/patternfly-elements/issues/2117

### CODE REVIEW

File: elements/pfe-clipboard-copy/pfe-clipboard-copy.scss [hunk 1]
```diff
@@ -1,3 +1,272 @@
+:root {
```
Comment (bennypowers) [inline elements/pfe-clipboard-copy/pfe-clipboard-copy.scss:1]: we shouldn't need to pull all this in. better to fall back

File: elements/pfe-clipboard-copy/pfe-clipboard-copy.ts [hunk 2]
```diff
@@ -20,83 +28,61 @@ export class PfeClipboardCopy extends LitElement {
...
+        'pf-m-expanded': this.expanded,
+        'pf-m-inline': this.variant === 'inline-compact',
+        'pf-m-block': this.block
+        })}
+      >
+        ${this.variant === 'inline-compact' ?
+          html`
+            ${this.code ?
+              html`
+                <code class="pf-c-clipboard-copy__text pf-m-code">
```
Comment (bennypowers) [inline elements/pfe-clipboard-copy/pfe-clipboard-copy.ts:58]: we'd shouldn't use BEM in shadow DOM

File: elements/pfe-clipboard-copy/pfe-clipboard-copy.scss (hunk 1 above)
Comment (heyMP) [reply elements/pfe-clipboard-copy/pfe-clipboard-copy.scss]: I put that stuff in there just to play around with different workflows for doing 1:1 conversion.  After seeing some other examples I'm starting with the plain BaseClass for functionality and will add the PF styles later.

File: elements/pfe-clipboard-copy/pfe-clipboard-copy.ts (hunk 2 above)
Comment (heyMP) [reply elements/pfe-clipboard-copy/pfe-clipboard-copy.ts]: Roger 👍

**Review by bennypowers — APPROVED** [review]
Great work, @kylebuch8, @heyMP, and @nikkimk!
 
I simplified the shadow template, removed a bunch of javascript, employed "shadow classes" pattern for managing state, added demos, and refactored test.

Well done, all. Looping Generates Tremendous Magnetism

**Pull request merged**